smolyay/surrogate.py

```diff
@@ -1,546 +1,717 @@
+import abc
+import itertools
+import warnings
+
 import numpy
+import sklearn
+import sklearn.linear_model
```

Comment:
Is it sufficient to import only the `linear_model` module?
```suggestion
import sklearn.linear_model
```